jetstream/README.md

```diff
@@ -603,6 +609,193 @@ ackF, err = js.PublishAsync("ORDERS.new", []byte("hello"))
+ctx := context.Background()
+
+// Create a new bucket. Bucket name is required and has to be unique within a stream.
+kv, _ := js.CreateKeyValue(ctx, jetstream.KeyValueConfig{Bucket: "profiles"})
+
+// Set a value for a given key
+// Put will either create or update a value for a given key
+kv.Put(ctx, "sue.color", []byte("blue"))
+
+// Get an entry for a given key
+// Entry contains key/value, but also metadata (revision, timestamp, etc.)) 
+entry, _ := kv.Get(ctx, "sue.color")
+
+// Prints `sue.color @ 1 -> "blue"`
+fmt.Printf("%s @ %d -> %q\n", entry.Key(), entry.Revision(), string(entry.Value()))
+
+// Update a value for a given key
+// Update will fail if the key does not exist or the revision is not up to date
+kv.Update(ctx, "sue.color", []byte("red"), 1)
+
+// Create will fail if the key already exists
+_, err := kv.Create(ctx, "sue.color", []byte("purple"))
+fmt.Println(err) // prints `nats: key exists`
+
+// Delete a value for a given key
+kv.Delete(ctx, "sue.color")
+
+// getting a deleted key will return an error
+_, err = kv.Get(ctx, "sue.color")
+fmt.Println(err) // prints `nats: key not found`
+
+// A bucket can be deleted once it is no longer needed
+js.DeleteKeyValue(ctx, "profiles")
+```
+
+### Watching for changes on a bucket
+
+KV buckets support Watchers, which can be used to watch for changes on a given
+key or the whole bucket. Watcher will receive a notification on a channel when a
+change occurs. By default, watcher will return initial values for all matching
+keys. After sending all initial values, watcher will send nil on the channel to
+signal that all initial values have been sent and it will start sending updates when
+changes occur.
+
+Watcher supports several configuration options:
+
+- `IncludeHistory` will have the key watcher send all historical values
+for each key (up to KeyValueMaxHistory).
+- `IgnoreDeletes` will have the key watcher not pass any keys with
+delete markers.
+- `UpdatesOnly` will have the key watcher only pass updates on values
+(without latest values when started).
+- `MetaOnly` will have the key watcher retrieve only the entry meta
+data, not the entry value.
+- `ResumeFromRevision` instructs the key watcher to resume from a
+specific revision number.
+
+```go
+js, _ := jetstream.New(nc)
+ctx := context.Background()
+kv, _ := js.CreateKeyValue(ctx, jetstream.KeyValueConfig{Bucket: "profiles"})
+
+kv.Put(ctx, "sue.color", []byte("blue"))
+
+// A watcher can be created to watch for changes on a given key or the whole bucket
+// Watcher will receive a notification on a channel when a change occurs
+// By default, watcher will return initial values for all matching keys.
+// Watcher can be configured to only return updates by using jetstream.UpdatesOnly() option.
+watcher, _ := kv.Watch(ctx, "sue.*")
+defer watcher.Stop()
+
+kv.Put(ctx, "sue.age", []byte("43"))
+kv.Put(ctx, "sue.color", []byte("red"))
+
+// First, the watcher sends initial values for all matching keys
```

Comment:
```suggestion
// First, the watcher sends last known values for all matching keys
```

Reply:
little different naming idea than the previous one

Reply:
I think I prefer "most recent", "last known" sounds a little vague to me.